Add PathPrefix to the generated TwirpServer interface.

gorzell opened the pull request:
*Issue:* #158

*Description of changes:*
The generated code implements the new function by returning the existing
const for this value. Having an interface method provides consistent
access for all TwirpServer instances, rather than having to use a
instance specific constant.

**Note:** If the `Dockerfile` isn't generally useful, I am happy to revert that commit.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

spenczar commented on protoc-gen-twirp/generator.go:
```diff
@@ -352,6 +352,10 @@ func (t *twirp) generateUtils() {
 	t.P(`	// ProtocGenTwirpVersion is the semantic version string of the version of`)
 	t.P(`	// twirp used to generate this file.`)
 	t.P(`	ProtocGenTwirpVersion() string`)
+	t.P(`// PathPrefix is used for all URL paths on the generated server.`)
```
I'm picky about comments. Could you phrase this as follows:

```go
// PathPrefix returns the HTTP URL path prefix for all methods handled by this
// service. This can be used with an HTTP mux to route twirp requests
// alongside non-twirp requests on one HTTP listener.
```

I realize that you've copied the comment that's on the `const`, but I think that comment could be improved too.

gorzell replied:
I don't mind changing this.  Should I change the comment on the `const` as well?

spenczar commented on Dockerfile:
```diff
@@ -0,0 +1,10 @@
+FROM golang:1.12
```
Should we use this in our travis builds, too?

gorzell replied:
If you mean the GoLang version, I think it is always nice to be current.  If you mean the container itself, I don't see why not, but I also don't know that much about how travis works.

spenczar replied:
I meant the container itself.

I think it's usually important that the way we do dev work mirrors exactly the way we do CI.

Could you revert the commit that adds the Docker container? I think it might be a good idea, but I'd like to do it more comprehensively, covering each of the Go versions we want to support, that sort of thing.

gorzell commented:
@spenczar I think I covered everything in: https://github.com/twitchtv/twirp/pull/167#pullrequestreview-219097865, and I reverted the `Dockerfile`.

I think the only open items at this point are:
* Was I too lazy/literal in the way I changed the Mux test?
* Should I change the `const` godoc to match the `PathPrefix()` version?

spenczar commented on example/service.twirp.go:
```diff
@@ -363,6 +367,10 @@ type TwirpServer interface {
 	// ProtocGenTwirpVersion is the semantic version string of the version of
 	// twirp used to generate this file.
 	ProtocGenTwirpVersion() string
+	// PathPrefix is used for all URL paths on the generated server.
```
Please re-run code generation with the new comment.

spenczar commented:
> Was I too lazy/literal in the way I changed the Mux test?
No, I really like the way you've done it.

> Should I change the const `godoc` to match the `PathPrefix()` version?
I don't think that's necessary in this revision.

gorzell commented:
@spenczar I think this should be ready to be merged now.

spenczar commented:
Thanks for the nice design, @gorzell. I've cut a new release, v5.6.0, which adds this feature, and I've updated the documentation website.
